### ISSUE TITLE
On change event is triggered 2 times on value change

### ISSUE THREAD

**Issue by Cristian-I** [Pretius/Pretius-APEX-Enhanced-Lov-Item#42]
Hey,

The change event is triggered 2 times on a value change of an enh lov field. 

How to reproduce:
Create a dynamic action with Event: Change, Item: the Enh LOV item. On true action add an Alert.

When you change the value from blank to some value the alert with be triggered just once but when you change the value again from a value to another the alert will be triggered 2 times. 
Same behavior on jquery .change() method.

Thank you,
Cristian

**Comment by bostrowski** [comment]
@Cristian-I 
Thank you for your feedback. I was able to reproduce the issue.

For the sake for future feedbacks please provide more information - for example the issue occurs only in autocomplete mode and when multiple selection is disabled. 

The best approach is to recreate the use-case scenario at apex.oracle.com.

I will get back to you with the fix.

**Comment by bostrowski** [comment]
@Cristian-I 
Check the latest commit to branch v1.1.1 and let me know whether the issue still occurs.

Download branch as zip file and install package / plugin installation file.

**Comment by Cristian-I** [comment]
@bostrowski Thank you for looking into this and my apologies for not providing an use-case scenario at apex.oracle.com

The change works and the `Change` event doesn't trigger 2 times. A bit strange is with the point it triggers now. 

From what i can see when it is autocomplete with single selection function `_promptSelectValue` calls first `_promptUnselectValue` and then `_promptApplyValues` both of witch can trigger the `change` event and with this fix you made it so that the `_promptApplyValues` will not trigger the event.
 
In my opinion it's better if `_promptUnselectValue` doesn't trigger the event and `_promptApplyValues` does. Because usually i need to catch the `change` event so that i run something after the value has changed. If it is triggered after `_promptUnselectValue` the field doesn't have a value yet and i have to wait for `_promptApplyValues` to have it.

And talking about getting the new value after the `change` event I've added to `apex.event.trigger( this.element.get(0), 'change')` the `this.mask.state.selected` as an addition argument so that i can capture and use it in DA or in js.

I've attached the changes (marked with `//double event change`) that i propose. I would love to hear your thoughts on them.
[enhancedLovItem_double_event.zip](https://github.com/Pretius/Pretius-APEX-Enhanced-Lov-Item/files/4117202/enhancedLovItem_double_event.zip)
